src/references/game_events.md

```diff
@@ -168,6 +170,231 @@ The `eventID` value can be retrieved by copying the **Game Event Id** for the ev
+    eventEndTime = eventData:GetEndDateTime().secondsSinceEpoch
+end
+
+function CheckEvent()
+    if eventData ~= nil then
+        --Submit all players if current time is between the start and end of an event
+        local currentTime = DateTime.CurrentTime().secondsSinceEpoch
+        if currentTime >= eventStartTime then
+            if currentTime < eventEndTime then
+                local allPlayers = Game.GetPlayers()
+                for _, player in pairs(allPlayers) do
+                    SubmitPlayer(player, currentTime)
+                end
+            else
+                StopTask()
+            end
+        end
+    else
+        StopTask()
+    end
+end
+
+--Stop the task once the event has ended or is invalid
+function StopTask()
+    checkEventTask:Cancel()
+    checkEventTask = nil
+end
+
+function SubmitPlayer(player, time)
+    if Leaderboards.HasLeaderboards() then
+        Leaderboards.SubmitPlayerScore(LEADERBOARD, player, time)
+    end
+end
+
+--Task repeats indefinitely every 10 seconds
+checkEventTask = Task.Spawn(CheckEvent)
+checkEventTask.repeatCount = -1
+checkEventTask.repeatInterval = 10
+
+--Need to check when a player joins in case the event ends before the task runs again
+function OnPlayerJoined(player)
+    if eventData ~= nil then
+        local currentTime = DateTime.CurrentTime().secondsSinceEpoch
+        if currentTime >= eventStartTime and currentTime < eventEndTime then
+            SubmitPlayer(player, currentTime)
+        end
+    end
+end
+
+Game.playerJoinedEvent:Connect(OnPlayerJoined)
+
+--Only the creator can print all participants by typing "/event" into chat
+function PrintEventParticipants(player, data)
+    if player.name == creatorName and data.message == "/event" then
+        if Leaderboards.HasLeaderboards() then
+            print("Printing Event Participants...")
+            local entries = Leaderboards.GetLeaderboard(LEADERBOARD, LeaderboardType.GLOBAL)
+            for _, entry in pairs(entries) do
+                print(entry.name)
+            end
+        end
+    end
+end
+
+Chat.receiveMessageHook:Connect(PrintEventParticipants)
+```
+
+<div class="mt-video" style="width:100%">
+    <video autoplay muted playsinline controls loop class="center" style="width:100%">
+        <source src="/img/GameEvents/leaderboard_example.mp4" type="video/mp4" />
+    </video>
+</div>
+
+### Spawning Event Template
+
+Events usually have items or activities that are temporary while the event is active such as a special boss fight. The start and end time of an event can be used to spawn a template if an event is active and destroy it once it has ended.
+
+```lua
+local TEMPLATE = script:GetCustomProperty("EventTemplate")
+--Parent object is at the desired location for the template
+local PARENT_OBJECT = script:GetCustomProperty("ParentObject"):WaitForObject()
+local EVENT_ID = script:GetCustomProperty("EventID")
+
+local eventData = CorePlatform.GetGameEvent(EVENT_ID)
```

Comment:
Comment above that this will yield. It's usually not long, but it will halt execution.